src/mono/wasm/Wasm.Build.Tests/BrowserStructures/RunOptions.cs

```diff
@@ -8,21 +8,27 @@
 
 #nullable enable
 
-namespace Wasm.Build.Tests.Blazor;
-public record BlazorRunOptions
+namespace Wasm.Build.Tests;
+public record RunOptions
 (
-    BlazorRunHost Host = BlazorRunHost.DotnetRun,
+    string Configuration,
+    RunHost Host = RunHost.DotnetRun,
     bool DetectRuntimeFailures = true,
     bool CheckCounter = true,
     Dictionary<string, string>? ServerEnvironment = null,
     Func<IPage, Task>? Test = null,
-    Action<IPage, IConsoleMessage>? OnConsoleMessage = null,
+    string? TestScenario = null,
```

Comment:
idea: This parameter is specific to WasmBasicTestApp. I'm not sure whether to have it here. Could we do it in a way that selected TestAsset defines what parameters we have?

Reply:
This is not the only one. We also have only-blazor: `CheckCounter`, `Test`.

I see a few options:
- Go back to 2 separate `RunOptions` inheriting from base class. This won't become that messy over time as having them without inheritance like we used to.
- Fixate the scenario in the phase of creating the app - use `UpdateFile` on `main.js` in each test that uses scenarios. That shifts the responsibility to test code, to decrease the effort we can make `DotnetRun` the default scenario if `test` is not passed.
- Remove scenarios and convert main.js to asset in different forms. Then we will copy corresponding `main.js` for specific scenario and no `TestScenario` is needed.